test/reduce_test.cpp

```diff
@@ -344,6 +347,22 @@ struct verify_reduce_with_indices
         std::size_t ws_sizeInBytes      = workspace.desc.GetElementSize() * sizeof(T);
         std::size_t indices_sizeInBytes = indices.desc.GetElementSize() * sizeof(int);
 
+        double alphaData, betaData;
+
+        void* alphaPara = reinterpret_cast<void*>(&alphaData);
+        void* betaPara  = reinterpret_cast<void*>(&betaData);
+
+        if(std::is_same<T, double>::value)
+        {
+            *reinterpret_cast<double*>(alphaPara) = static_cast<double>(alpha);
+            *reinterpret_cast<double*>(betaPara)  = static_cast<double>(beta);
+        }
+        else
+        {
+            *reinterpret_cast<float*>(alphaPara) = alpha;
+            *reinterpret_cast<float*>(betaPara)  = beta;
+        };
+
```

Comment:
```suggestion
        const double alpha64 = alpha;
        const double beta64 = beta;
        const void* const alphaPtr = (std::is_same<T, double>::value) ? &alpha64 : &alpha;
        const void* const betaPtr = (std::is_same<T, double>::value) ? &beta64 : &beta;
```
Then pass alphaPtr and betaPtr.

If compiler or tidy will ask for casting (basically it shouldn't), then use static cast:
```C++
        const void* const alphaPtr = (std::is_same<T, double>::value) ? static_cast<const void*>(&alpha64)...
```

Reply:
Yes.